File: Ejercicio 018/sistemaUni.py

class Persona:

    def __init__(self, identificacion, nombre, contacto):
        self.identificacion = identificacion
        self.nombre = nombre
        self.contacto = contacto

    def mostrar_datos(self):
        print(f"{self.nombre} | ID: {self.identificacion} | Contacto: {self.contacto}")



class Estudiante(Persona):

    def __init__(self, id, nombre, contacto, programa, semestre):
        super().__init__(id, nombre, contacto)
        self.programa = programa
        self.semestre = semestre

    def estudiar(self):
        print(f"{self.nombre} estudia {self.programa} semestre {self.semestre}")


class Docente(Persona):

    def __init__(self, id, nombre, contacto, area, materia):
        super().__init__(id, nombre, contacto)
        self.area = area
        self.materia = materia

    def ensenar(self):
        print(f"{self.nombre} dicta {self.materia} en {self.area}")


class Administrativo(Persona):

    def __init__(self, id, nombre, contacto, cargo):
        super().__init__(id, nombre, contacto)
        self.cargo = cargo

    def gestionar(self):
        print(f"{self.nombre} trabaja como {self.cargo}")


e = Estudiante("101","Ana","ana@uni.edu","Ingeniería",5)
d = Docente("202","Carlos","carlos@uni.edu","Sistemas","POO")
a = Administrativo("303","Laura","laura@uni.edu","Secretaria")

e.mostrar_datos()
e.estudiar()

d.mostrar_datos()
d.ensenar()

a.mostrar_datos()
a.gestionar()

File: Ejercicio 018/sistemaUni.java
class Persona {

    protected String identificacion;
    protected String nombre;
    protected String contacto;

    public Persona(String id, String nombre, String contacto) {
        this.identificacion = id;
        this.nombre = nombre;
        this.contacto = contacto;
    }

    public void mostrarDatos() {
        System.out.println(nombre + " | ID: " + identificacion +
                " | Contacto: " + contacto);
    }
}



class Estudiante extends Persona {

    private String programa;
    private int semestre;

    public Estudiante(String id, String nombre, String contacto,
                      String programa, int semestre) {
        super(id, nombre, contacto);
        this.programa = programa;
        this.semestre = semestre;
    }

    public void estudiar() {
        System.out.println(nombre + " estudia " + programa +
                " semestre " + semestre);
    }
}

class Docente extends Persona {

    private String area;
    private String materia;

    public Docente(String id, String nombre, String contacto,
                   String area, String materia) {
        super(id, nombre, contacto);
        this.area = area;
        this.materia = materia;
    }

    public void enseñar() {
        System.out.println(nombre + " dicta " + materia +
                " en el área " + area);
    }
}

class Administrativo extends Persona {

    private String cargo;

    public Administrativo(String id, String nombre, String contacto,
                          String cargo) {
        super(id, nombre, contacto);
        this.cargo = cargo;
    }

    public void gestionar() {
        System.out.println(nombre + " trabaja como " + cargo);
    }
}

public class Main {
    public static void main(String[] args) {

        Estudiante e = new Estudiante("101","Ana","ana@uni.edu","Ingeniería",5);
        Docente d = new Docente("202","Carlos","carlos@uni.edu","Sistemas","POO");
        Administrativo a = new Administrativo("303","Laura","laura@uni.edu","Secretaria");

        e.mostrarDatos();
        e.estudiar();

        d.mostrarDatos();
        d.enseñar();

        a.mostrarDatos();
        a.gestionar();
    }
}
